tests/unit/paginator.test.js

```diff
@@ -1,14 +1,76 @@
 const appRoot = require('app-root-path');
 const { assert } = require('chai');
+const _ = require('lodash');
 
 const { paginate } = appRoot.require('/utils/paginator');
 const rows = appRoot.require('/tests/unit/mock-data.json').pets;
 
+const repeatForAllPages = (assertion, assertionVars = {}) => {
+  const page = { size: 4, number: 1 };
+  const { totalPages } = paginate(rows, page);
+  assertionVars.totalPages = totalPages;
+  for (; page.number <= totalPages; page.number += 1) {
+    assertionVars.page = page;
+    assertion(assertionVars);
+  }
+};
+
 describe('Test paginator', () => {
-  it('number of returned results should less then page size', (done) => {
-    const page = { size: 10 };
-    const { paginatedRows } = paginate(rows, page);
-    assert.isAtMost(paginatedRows.length, page.size);
+  it('number of returned results is at most page size', (done) => {
+    const assertResultsSize = ({ page }) => {
+      const { paginatedRows } = paginate(rows, page);
+      assert.isAtMost(paginatedRows.length, page.size);
+    };
+    repeatForAllPages(assertResultsSize);
+    done();
+  });
+
+  it('unique IDs do not appear more than once across all pages', (done) => {
+    const assertNoDuplicateResults = (assertionVars) => {
+      const { paginatedRows } = paginate(rows, assertionVars.page);
+      _.forEach(paginatedRows, ({ ID }) => {
+        assert.isFalse(assertionVars.uIDs.includes(ID));
```

Comment:
Base on the [naming convention from our style guide](https://github.com/osu-mist/style-guides/blob/master/JavaScript.md#naming-conventions), acronyms should not be capitalized.